test/refactoring/ConvertAsync/convert_pattern.swift

```diff
@@ -5,6 +5,412 @@ enum E : Error { case e }
+    guard var y = strs else { return }
+    guard var z = strs else { return }
+    print(x, y, z)
+  }
+
+  // ALIAS-BINDINGS3:      var x = try await stringTupleParam()
+  // ALIAS-BINDINGS3-NEXT: var y = x
+  // ALIAS-BINDINGS3-NEXT: var z = x
+  // ALIAS-BINDINGS3-NEXT: print(x, y, z)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=ALIAS-BINDINGS4 %s
+  stringTupleParam { strs, err in
+    guard var x = strs else { return }
+    guard let y = strs else { return }
+    guard let z = strs else { return }
+    print(x, y, z)
+  }
+
+  // ALIAS-BINDINGS4:      let y = try await stringTupleParam()
+  // ALIAS-BINDINGS4-NEXT: var x = y
+  // ALIAS-BINDINGS4-NEXT: print(x, y, y)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=ALIAS-BINDINGS5 %s
+  stringTupleParam { strs, err in
+    guard var x = strs else { return }
+    print(x)
+  }
+
+  // ALIAS-BINDINGS5:      var x = try await stringTupleParam()
+  // ALIAS-BINDINGS5-NEXT: print(x)
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=STRING-TUPLE-RESULT %s
+  stringTupleResult { res in
+    switch res {
+    case .success((let x, let y)):
+      print(x, y)
+    case .failure:
+      print("oh no")
+    }
+  }
+  // STRING-TUPLE-RESULT:      do {
+  // STRING-TUPLE-RESULT-NEXT:   let (x, y) = try await stringTupleResult()
+  // STRING-TUPLE-RESULT-NEXT:   print(x, y)
+  // STRING-TUPLE-RESULT-NEXT: } catch {
+  // STRING-TUPLE-RESULT-NEXT:   print("oh no")
+  // STRING-TUPLE-RESULT-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MIXED-TUPLE-RESULT %s
+  mixedTupleResult { res in
+    if case .failure(let err) = res {
+      print("oh no")
+    }
+    if case .success(((let x, let y), let z)) = res {
+      print("a", x, y, z)
+    }
+    switch res {
+    case .success(let ((x, _), z)):
+      print("b", x, z)
+    case .failure:
+      print("oh no again")
+    }
+  }
+
+  // MIXED-TUPLE-RESULT:      do {
+  // MIXED-TUPLE-RESULT-NEXT:   let res = try await mixedTupleResult()
+  // MIXED-TUPLE-RESULT-NEXT:   let ((x, y), z) = res
+  // MIXED-TUPLE-RESULT-NEXT:   let ((x1, _), z1) = res
+  // MIXED-TUPLE-RESULT-NEXT:   print("a", x, y, z)
+  // MIXED-TUPLE-RESULT-NEXT:   print("b", x, z)
+  // MIXED-TUPLE-RESULT-NEXT: } catch let err {
+  // MIXED-TUPLE-RESULT-NEXT:   print("oh no")
+  // MIXED-TUPLE-RESULT-NEXT:   print("oh no again")
+  // MIXED-TUPLE-RESULT-NEXT: }
+
+  // RUN: %refactor-check-compiles -convert-call-to-async-alternative -dump-text -source-filename %s -pos=%(line+1):3 | %FileCheck -check-prefix=MIXED-TUPLE-RESULT2 %s
+  mixedTupleResult { res in
```

Comment:
There's probably a few tests in here that are a little superfluous given others but... eh.

At some point I'd like to update the %refactor* tools to be similar to the swift-ide-test batch completion ones given the number we're adding 😆.

Reply:
Ah, I think what happened here is I ended up adding the `MIXED-TUPLE-RESULT` test case after this one and then didn't notice the overlap. I'll tweak this one so it's at least testing something slightly different.